data-hibernate-jpa/src/main/java/io/micronaut/data/hibernate/operations/AbstractHibernateOperations.java

```diff
@@ -470,7 +470,7 @@ public Object convert(Class<?> converterClass, Object value, Argument<?> argumen
             @Override
             public void bindOne(QueryParameterBinding binding, Object value) {
                 String parameterName = Objects.requireNonNull(binding.getName(), "Parameter name cannot be null!");
-                if (storedQuery.isNative()) {
+                if (binding.getParameterIndex() != -1) {
```

Comment:
@rawilder Does it work without the native check change?

Reply:
Then it goes into else branch 
```
else if (bindNamed) {
                    setParameter(q, parameterName, value);
                } else {
                    setParameter(q, index, value);
                }
```
and since it was an empty collection, in binding method I posted above it is being set to null. And in this else branch, we don't have pointer to the argument type, don't know if this is collection or not since it is null and we set null parameter to Hibernate and then it fails here
```
@Override
	public boolean isInstance(Object value) {
		final LazyInitializer lazyInitializer = extractLazyInitializer( value );
		final Class<T> javaTypeClass = getJavaTypeClass();
		if ( lazyInitializer != null ) {
			return javaTypeClass.isAssignableFrom( lazyInitializer.getPersistentClass() )
					|| javaTypeClass.isAssignableFrom( lazyInitializer.getImplementationClass() );
		}
		else {
			return javaTypeClass.isAssignableFrom( value.getClass() );
		}
	}
```
on line `return javaTypeClass.isAssignableFrom( value.getClass() );` because `value` is null throws NPE.
For difference, code in `if` branch tries to convert null parameter to an empty collection is recognizes it is collection argument.

Reply:
From what I can see by debugging code with this change, this branch
```
else if (bindNamed) {
                    setParameter(q, parameterName, value);
                }
```
is used only in save, update, delete methods.

Reply:
Any thoughts on the above @dstepanov ?

Reply:
Can you please target 4.9

Reply:
Retargeted. I changed this line back to `isNative` and reran the tests for good measure and the one I added failed, so I changed it back to the `getParameterIndex` condition.

Reply:
Okay monkeyed with the branch quite a bit to get it to be just the work's commits again, branched off of 4.9.x, should be the same changes (minus the accidentally included imports)